src/main/java/com/exadel/discount/repository/impl/DiscountCustomRepositoryImpl.java

```diff
@@ -0,0 +1,33 @@
+package com.exadel.discount.repository.impl;
+
+import com.exadel.discount.entity.QDiscount;
+import com.exadel.discount.repository.DiscountCustomRepository;
+import com.querydsl.core.types.Predicate;
+import com.querydsl.jpa.impl.JPAQueryFactory;
+import org.springframework.data.domain.Pageable;
+
+import javax.persistence.EntityManager;
+import java.util.List;
+import java.util.UUID;
+
+public class DiscountCustomRepositoryImpl implements DiscountCustomRepository {
```

Comment:
I don't see annotations. Who and how creates this class and pass em to constructor?

Reply:
+